source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,170 @@
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include <array>
+#include <string>
+
+#include "envoy/common/time.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// Utility functions used in RequestCacheControl & ResponseCacheControl
+namespace {
+OptionalDuration parseDuration(absl::string_view s) {
```

Comment:
suggest making these less private so you can unit-test them directly.

Reply:
For now, they are only used by and tested through RequestCacheControl and ResponseCacheControl, that's why they are private to this file.

Reply:
@toddmgreer what do you think?

Reply:
Seems like it would be easier to thoroughly test this on its own.

Reply:
I'm not a fan of making helpers public for testing when (as is the case here) we can adequately test the public API instead. However, I'll support whatever you work out with @jmarantz.

Reply:
@jmarantz I am happy to add tests for the function, my point is just that it is fairly straight forward and is already tested through existing tests for RequestCacheControl & ResponseCacheControl.

Reply:
I believe the positive cases are all tested. I was thinking of testing all kinds of malformed durations. Are they all tested? They might be but it's harder to tell cause there's no focused testing at this layer.

Reply:
The only logic this function performs is removing any existing quotes (which is tested). The actual parsing and detection of malformed digits is actually done through `absl::SimpleAtoi`. If it parses a valid number, it is used as the duration, otherwise the duration is ignored.

Reply:
Some malformed durations are tested through `RequestCacheControl` and `ResponseCacheControl`, but not all kinds of malformed durations as this would really be testing of `absl::SimpleAtoi`.